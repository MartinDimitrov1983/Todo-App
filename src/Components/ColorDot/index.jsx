import React from 'react';
import PropTypes from 'prop-types';
import styles from './index.module.css';

const ColorDot = ({ color, onClick }) => {
  return (
    <div
      className={styles.dot}
      style={{ background: color }}
      onClick={() => onClick()}
    ></div>
  );
};

ColorDot.propTypes = {
  color: PropTypes.string,
  onClick: PropTypes.func,
};

export default ColorDot;
