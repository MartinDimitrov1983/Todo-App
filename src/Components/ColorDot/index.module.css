.dot {
  width: 15px;
  height: 15px;
  border-radius: 50%;
  cursor: pointer;
  border: 1px solid black;
}
