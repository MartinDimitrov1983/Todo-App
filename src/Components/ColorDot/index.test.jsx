import React from 'react';
import { render } from '@testing-library/react';

import ColorDot from './index';

test('Snapshot Dot', () => {
  const { asFragment } = render(
    <ColorDot color="#ffffff" onClick={() => {}} />,
  );
  expect(asFragment()).toMatchSnapshot();
});
