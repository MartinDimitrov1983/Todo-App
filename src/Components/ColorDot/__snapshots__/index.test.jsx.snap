// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Dot 1`] = `
<DocumentFragment>
  <div
    class="dot"
    style="background: rgb(255, 255, 255);"
  />
</DocumentFragment>
`;
